src/sage/modular/modform_hecketriangle/space.py

```diff
@@ -482,29 +481,29 @@ def coordinate_vector(self, v):
             sage: el2 == MF.element_from_coordinates(el2.coordinate_vector())
             True
         """
-
-        (x,y,z,d) = self.pol_ring().gens()
+        x, y, z, d = self.pol_ring().gens()
         k = self._weight
-        rmax = QQ(k / ZZ(2)).floor()
+        rmax = k // 2
```

Comment:
This change is slightly dangerous to me (as much as I agree with it in principle). Are we sure about what the type of `k` should be; mainly, an `int` or an `Integer`?

Reply:
indeed, k seems to be a rational. Changed to something safer now.